railties/lib/rails/test_unit/test_requirer.rb

```diff
@@ -15,7 +15,7 @@ def require_files(patterns)
       private
         def expand_patterns(patterns)
           patterns.map do |arg|
-            arg = arg.gsub(/:(\d+)?$/, '')
+            arg = arg.gsub(/(:\d*)+?$/, '')
```

Comment:
Changed it in https://github.com/rails/rails/commit/88881d2b50b5d89fc30c77c2e1669b858bbf418a — thanks for catching it all.